controlplane/kubeadm/internal/controllers/upgrade.go

```diff
@@ -74,8 +74,20 @@ func (r *KubeadmControlPlaneReconciler) upgradeControlPlane(
 		return ctrl.Result{}, errors.Wrap(err, "failed to update the kubernetes version in the kubeadm config map")
 	}
 
+	var imageRepository string
 	if kcp.Spec.KubeadmConfigSpec.ClusterConfiguration != nil {
-		imageRepository := kcp.Spec.KubeadmConfigSpec.ClusterConfiguration.ImageRepository
+		imageRepository = kcp.Spec.KubeadmConfigSpec.ClusterConfiguration.ImageRepository
+	}
+
+	// Beginning with kubernetes v1.25, kubernetes images get published to registry.k8s.io instead of k8s.gcr.io.
+	// This ensures that the imageRepository gets patched to registry.k8s.io when upgrading from v1.24 or lower,
+	// but only if there was no imageRespository explicitly set at the KubeadmControlPlanes ClusterConfiguration.
+	// This follows the behavior of `kubeadm upgrade`.
+	if imageRepository == "" && parsedVersion.GTE(semver.Version{Major: 1, Minor: 25, Patch: 0, Pre: []semver.PRVersion{{VersionStr: "alpha"}}}) {
```

Comment:
Could you kindly introduce a const like e.g. minVerUnversionedKubeletConfig, also, not sure about the Pre part, we don't have it in other cases